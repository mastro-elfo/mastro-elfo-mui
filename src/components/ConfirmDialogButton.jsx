import React, { Fragment, useState } from "react";
import PropTypes from "prop-types";

import { Button, IconButton } from "@material-ui/core";

import ConfirmDialog from "./ConfirmDialog";

/**
 * Creates a `Button`  or an `IconButton` that opens a `ConfirmDialog`.
 * @param       {[type]}  children       [description]
 * @param       {Boolean} [isIcon=false] [description]
 * @param       {[type]}  [onConfirm=(]  [description]
 * @constructor
 */
export default function ConfirmDialogButton({
  children,
  isIcon = false,
  onConfirm = () => {},
  DialogProps = {},
  ...others
}) {
  const [open, setOpen] = useState(false);

  // Use Button or IconButton
  const Component = isIcon ? IconButton : Button;

  const handleConfirm = () => {
    setOpen(false);
    onConfirm();
  };

  return (
    <Fragment>
      <Component color="inherit" onClick={() => setOpen(true)} {...others}>
        {children}
      </Component>
      <ConfirmDialog
        {...DialogProps}
        open={open}
        onClose={() => setOpen(false)}
        onCancel={() => setOpen(false)}
        onConfirm={handleConfirm}
      />
    </Fragment>
  );
}

ConfirmDialogButton.propTypes = {
  children: PropTypes.node.isRequired,
  isIcon: PropTypes.bool,
  onConfirm: PropTypes.func,
  ButtonProps: PropTypes.object,
  DialogProps: PropTypes.object
};
