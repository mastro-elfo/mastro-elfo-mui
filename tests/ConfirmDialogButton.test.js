import React from "react";
import renderer from "react-test-renderer";

import ConfirmDialogButton from "../src/components/ConfirmDialogButton";

function Custom({ children, ...props }) {
  return <div {...props}>{children}</div>;
}

test("Render Component", () => {
  const component = renderer.create(
    <ConfirmDialogButton Component={Custom} data-custom={true}>
      <span />
    </ConfirmDialogButton>
  );
  const tree = component.toJSON();
  expect(tree).toBeTruthy();
});
