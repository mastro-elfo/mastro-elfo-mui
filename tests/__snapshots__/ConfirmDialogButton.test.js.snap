// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component 1`] = `
<div
  data-custom={true}
  onClick={[Function]}
>
  <span />
</div>
`;
